File: QMP/QMPMOD.pas
unit
  QMPMOD;

interface

uses
  QMPDCL,
  QMPEQU,
  QMPENH;

type
  TQMPMOD = record
  private
    class function Plugin(const Which: Integer): PPlugin; cdecl; static;
  public
    class function Module(): PModule; cdecl; static;
  end;

implementation

class function TQMPMOD.Plugin(const Which: Integer): PPlugin; cdecl;
begin
  case (Which) of
    0: begin
      Result := TQMPEQU.Plugin();
    end;
    1: begin
      Result := TQMPENH.Plugin();
    end;
    else begin
      Result := nil;
    end;
  end;
end;

class function TQMPMOD.Module(): PModule; cdecl;
begin
  Result := New(PModule);
  Result.Plugin := TQMPMOD.Plugin;
end;

begin
end.

File: CPP/qmpmod.cpp
#pragma once
#include "qmpdcl.h"
#include "qmpmod.h"
#include "qmpequ.h"
#include "qmpenh.h"
#include "windows.h"

using namespace std;

PPLUGIN CDECL QMPMOD::plugin(INT which) {
    switch (which) {
    case 0:
        return QMPEQU::plugin();
        break;
    case 1:
        return QMPENH::plugin();
        break;
    default:
        return NULL;
        break;
    };

    return NULL;
};

PMODULE CDECL QMPMOD::module() {
    PMODULE result = new MODULE();

    result->plugin = QMPMOD::plugin;

    return result;
};
